tests/test_assets.py

```diff
@@ -1541,3 +1541,60 @@ def non_materializing_pipeline():
         match="Can only add metadata to assets that are arguments to @materialize",
     ):
         non_materializing_pipeline()
+
+
+@pytest.mark.usefixtures("reset_worker_events")
+def test_nested_materialization(asserting_events_worker: EventsWorker):
+    """Test nested materialization - a materialize task called inside another materialize task.
```

Comment:
do you think its worth adding a case where we call a materialize task from within a vanilla task?

Reply:
yes, added!